applications/registration-api/src/test/java/no/dcat/controller/ApiRegistrationControllerIT.java

```diff
@@ -0,0 +1,211 @@
+package no.dcat.controller;
+
+import com.google.gson.Gson;
+import no.dcat.datastore.ElasticDockerRule;
+import no.dcat.model.ApiRegistration;
+import no.dcat.model.Catalog;
+import no.dcat.service.ApiRegistrationRepository;
+import no.dcat.service.CatalogRepository;
+import no.dcat.shared.testcategories.IntegrationTest;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithUserDetails;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ActiveProfiles(value = "unit-integration")
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = RANDOM_PORT)
+@AutoConfigureMockMvc
+@Category(IntegrationTest.class)
+public class ApiRegistrationControllerIT {
+
+  @ClassRule public static ElasticDockerRule elasticRule = new ElasticDockerRule();
+  private static Logger logger = LoggerFactory.getLogger(ApiRegistrationControllerIT.class);
+  @Autowired private MockMvc mockMvc;
+  @Autowired private CatalogRepository catalogRepository;
+  @Autowired private ApiRegistrationRepository apiRegistrationRepository;
+  
+  public static String asJsonString(Object obj) {
+    return new Gson().toJson(obj);
+  }
+
+  @Before
+  public void before() {
+    try {
+      catalogRepository.deleteAll();
+    } catch (Exception e) {
+      logger.debug("catalogRepository was probably empty");
+    }
+
+    try {
+      apiRegistrationRepository.deleteAll();
+    } catch (Exception e) {
+      logger.debug("apiRegistrationRepository was probably empty");
+    }
+  }
+
+  @Test
+  @WithUserDetails("01066800187")
+  public void postApiRegistration() throws Exception {
+
+    Catalog catalog = new Catalog();
+    String catalogId = "910244999";
+    catalog.setId(catalogId);
+
+    mockMvc
```

Comment:
Du allerede har helper funksjon som lagrer katalog, hforfor ikke bruke dette.
Kanskje du skulle separere hjelpefunksion for katalog fra hjelpefunksjon som lagrer api.